src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,203 @@
+  });
+};
+
+const getCurrentChecksum = (params: SearchThresholdAlertParams) =>
+  sha256.create().update(JSON.stringify(params)).hex();
+
+const isActualAlert = (queryParams: QueryParams): queryParams is NonNullableEntry<QueryParams> => {
+  return Boolean(queryParams.from && queryParams.to && queryParams.checksum);
+};
+
+export function ViewAlertRoute() {
+  const { core, data, locator, toastNotifications } = useDiscoverServices();
+  const { id } = useParams<{ id: string }>();
+  const history = useHistory();
+  const { search } = useLocation();
+
+  const query = useMemo(() => new URLSearchParams(search), [search]);
+
+  const queryParams: QueryParams = useMemo(
+    () => ({
+      from: query.get('from'),
+      to: query.get('to'),
+      checksum: query.get('checksum'),
+    }),
+    [query]
+  );
+
+  const openConcreteAlert = isActualAlert(queryParams);
+
+  useEffect(() => {
+    const fetchAlert = async () => {
+      try {
+        return await core.http.get<Alert<SearchThresholdAlertParams>>(
+          `${LEGACY_BASE_ALERT_API_PATH}/alert/${id}`
+        );
+      } catch (error) {
+        const errorTitle = i18n.translate('discover.viewAlert.alertRuleFetchErrorTitle', {
+          defaultMessage: 'Error fetching alert rule',
+        });
+        displayError(errorTitle, error.message, toastNotifications);
+      }
+    };
+
+    const fetchSearchSource = async (fetchedAlert: Alert<SearchThresholdAlertParams>) => {
+      try {
+        return await data.search.searchSource.create(fetchedAlert.params.searchConfiguration);
+      } catch (error) {
+        const errorTitle = i18n.translate('discover.viewAlert.searchSourceErrorTitle', {
+          defaultMessage: 'Error fetching search source',
+        });
+        displayError(errorTitle, error.message, toastNotifications);
+      }
+    };
+
+    const showDataViewFetchError = (alertId: string) => {
+      const errorTitle = i18n.translate('discover.viewAlert.dataViewErrorTitle', {
+        defaultMessage: 'Error fetching data view',
+      });
+      displayError(
+        errorTitle,
+        new Error(`Data view failure of the alert rule with id ${alertId}.`).message,
+        toastNotifications
+      );
+    };
+
+    const navigateToResults = async () => {
+      const fetchedAlert = await fetchAlert();
+      if (!fetchedAlert) {
+        history.push(DISCOVER_MAIN_ROUTE);
+        return;
+      }
+
+      const calculatedChecksum = getCurrentChecksum(fetchedAlert.params);
+      if (openConcreteAlert && calculatedChecksum !== queryParams.checksum) {
+        displayRuleChangedWarn(toastNotifications);
+      } else if (openConcreteAlert && calculatedChecksum === queryParams.checksum) {
+        displayPossibleDocsDiffInfoAlert(toastNotifications);
+      }
+
+      const fetchedSearchSource = await fetchSearchSource(fetchedAlert);
+      if (!fetchedSearchSource) {
+        history.push(DISCOVER_MAIN_ROUTE);
```

Comment:
Should there be some sort of error warning here?

Reply:
Error case handled by `fetchSearchSource` function.